src/main/java/com/example/bookstoreapp/models/CatalogItem.java

```diff
@@ -10,20 +10,23 @@
 @NoArgsConstructor
 public class CatalogItem {
 
-  private Long id;
-  private String name;
-  private Float price;
-  private Integer stockQuantity;
+    private Long id;
+    private String name;
+    private Float price;
+    private Integer stockQuantity;
+    private Integer itemLimit;
 
-  public CatalogItemEntity toEntity() {
-    return new CatalogItemEntity(this.getId(), this.getName(), this.getPrice(), this.getStockQuantity());
-  }
+    public CatalogItemEntity toEntity() {
+        Integer itemLimits = (this.getItemLimit() == null) ? 0 : this.getItemLimit();
+        return new CatalogItemEntity(this.getId(), this.getName(), this.getPrice(), this.getStockQuantity(), itemLimits);
+    }
 
-  public CatalogItem fromEntity(CatalogItemEntity entity) {
-    this.setId(entity.getId());
-    this.setName(entity.getName());
-    this.setPrice(entity.getPrice());
-    this.setStockQuantity(entity.getStockQuantity());
-    return this;
-  }
+    public CatalogItem fromEntity(CatalogItemEntity entity) {
+        this.setId(entity.getId());
+        this.setName(entity.getName());
+        this.setPrice(entity.getPrice());
+        this.setStockQuantity(entity.getStockQuantity());
+        this.setItemLimit(entity.getItemLimit());
```

Comment:
item limit changes not required in this pull request, you can stash the commit to separate branch and create another pull requests for item limit.

Reply:
crated a separated pull request for the issue